v4/core/base_service.go

```diff
@@ -256,8 +268,25 @@ func (service *BaseService) Request(req *http.Request, result interface{}) (deta
 		return
 	}
 
-	// Invoke the request.
-	httpResponse, err := service.Client.Do(req)
+	var httpResponse *http.Response
+
+	// Try to get the retryable Client hidden inside service.Client
+	retryableClient := getRetryableHTTPClient(service.Client)
```

Comment:
If the BaseService.Client field is a "retryable" client, then it will be an instance of http.Client with an alternate transport impl which contains the entire "retryable" client instance.   This step here is to check if we in fact are using a retryable client and to return it (from the transport) if so.   We do this so we can invoke the retryable client differently from a normal client.